notebooks/4.visualization/confusion_matrix/confusion_matrix.r

```diff
@@ -0,0 +1,74 @@
+suppressPackageStartupMessages(suppressWarnings(library(lintr))) # linting
+suppressPackageStartupMessages(suppressWarnings(library(ggplot2))) # plotting
+suppressPackageStartupMessages(suppressWarnings(library(dplyr))) # data manipulation
+suppressPackageStartupMessages(suppressWarnings(library(reshape2))) # data manipulation
+suppressPackageStartupMessages(suppressWarnings(library(ggridges))) # ridgeline plots
+suppressPackageStartupMessages(suppressWarnings(library(RColorBrewer))) # color palettes
+
+# get path for confusion matrix
+confusion_matrix_path <- file.path("../../../results/2.modeling/confusion_matrix.csv.gz")
+overlapping_confusion_matrix_path <- file.path("../../../results/2.modeling/jump_overlap_confusion_matrix.csv.gz")
+cm_df <- read.csv(confusion_matrix_path, sep = ",")
+overlapping_cm_df <- read.csv(overlapping_confusion_matrix_path, sep = ",")
+
+# update one of the shuffled_model columns to Shuffled and Not Shuffled
+cm_df$shuffled_model <- ifelse(cm_df$shuffled_model == "False", "Not Shuffled",
+                            ifelse(cm_df$shuffled_model == "True", "Shuffled", cm_df$shuffled_model))
+
+# Display the updated DataFrame
+head(cm_df)
+
+# update one of the shuffled_model columns to Shuffled and Not Shuffled
+overlapping_cm_df$shuffled_model <- ifelse(overlapping_cm_df$shuffled_model == "False", "Not Shuffled",
+                            ifelse(overlapping_cm_df$shuffled_model == "True", "Shuffled", overlapping_cm_df$shuffled_model))
+
+# Display the updated DataFrame
+head(overlapping_cm_df)
+
+# set image dim
+width <- 10
+height <- 10
+options(repr.plot.width = width, repr.plot.height = height)
+
+# create a confusion matrix plot
+confusion_matrix_plot <- (
+    ggplot(cm_df, aes(x = true_label, y = predicted_labels))
+    + facet_grid(dataset_type~shuffled_model)
+    + geom_point(aes(color = recall), size = 3, shape = 15)
+    + geom_text(aes(label = count), size = 2)
+    + scale_color_gradient("Recall", low = "white", high = "red",limits = c(0, 1))
+    + theme_bw()
+    + ylab("True Class")
+    + xlab("Predicted Class")
+    + theme(strip.text = element_text(size = 12))
+    + theme(axis.text.x = element_text(angle = 90, hjust = 1))
+)
+
+# saving file
+ggsave(filename = "confusion_matrix.png", height = height, width = width)
```

Comment:
for publication readiness 
```suggestion
ggsave(filename = "confusion_matrix.png", height = height, width = width, dpi=600)
```